src/go/rucio-dataset-mon-go/controllers/controllers.go

```diff
@@ -0,0 +1,244 @@
+
+		// Try custom
+		rCustom := models.DataTableCustomRequest{}
+		errCustom := c.ShouldBindBodyWith(&rCustom, binding.JSON)
+		if errCustom == nil {
+			datasetsResp = datasets.GetResults(ctx, c, rCustom)
+			c.JSON(http.StatusOK,
+				datasetsResp,
+			)
+			verboseControllerOutLog(start, "GetDatasets", rCustom, datasetsResp)
+			return
+		}
+		tempReqBody, _ := c.Get(gin.BodyBytesKey)
+		log.Printf("[WARN] Request could not bind with custom DT request %#v", string(tempReqBody.([]byte)))
+
+		// Try plain
+		rPlain := models.DataTableRequest{}
+		errPlain := c.ShouldBindBodyWith(&rPlain, binding.JSON)
+		if errPlain == nil {
+			datasetsResp = datasets.GetResults(ctx, c, rCustom)
+			c.JSON(http.StatusOK,
+				datasetsResp,
+			)
+			verboseControllerOutLog(start, "GetDatasets", rPlain, datasetsResp)
+			return
+		}
+		tempReqBody, _ = c.Get(gin.BodyBytesKey)
+		log.Printf("[ERROR] Request could not bind with custom DT request %#v", string(tempReqBody.([]byte)))
+		utils.ErrorResponse(c, "Bad request", errCustom, string(tempReqBody.([]byte)))
+		return
+	}
+}
+
+// GetDetailedDs controller that returns datasets according to DataTable request json
+func GetDetailedDs() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		ctx, cancel, start, req := controllerInitialize(c)
+		defer cancel()
+		detailedDatasetsResp := detailed_datasets.GetResults(ctx, c, req)
+		c.JSON(http.StatusOK,
+			detailedDatasetsResp,
+		)
+		verboseControllerOutLog(start, "GetDetailedDs", req, detailedDatasetsResp)
+		return
+	}
+}
+
+// GetShortUrlParam controller that returns short url param which is md5 hash of the datatables request
+func GetShortUrlParam() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		log.SetFlags(log.LstdFlags | log.Lshortfile)
+		verboseControllerInitLog(c)
+		start := time.Now()
+		ctx, cancel := context.WithTimeout(context.Background(), mongo.Timeout)
+
+		// Get request json with validation
+		req := models.ShortUrlRequest{}
+		if err := c.ShouldBindBodyWith(&req, binding.JSON); err != nil {
+			tempReqBody, _ := c.Get(gin.BodyBytesKey)
+			utils.ErrorResponse(c, "Bad request", err, string(tempReqBody.([]byte)))
+		}
+		defer cancel()
+		requestHash := short_url.GetShortUrl(ctx, c, req)
+		c.JSON(http.StatusOK,
+			requestHash,
+		)
+		verboseControllerOutLog(start, "GetShortUrlParam", req, requestHash)
+		return
+	}
+}
+
+// GetIndexPageFromShortUrlId controller that returns page from short url hash id
+func GetIndexPageFromShortUrlId() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		verboseControllerInitLog(c)
+		start := time.Now()
+		ctx, cancel := context.WithTimeout(context.Background(), mongo.Timeout)
+		defer cancel()
+
+		hashId := c.Param("id")
+		log.Printf("[INFO] Hash Id: %s", hashId)
+		shortUrlObj := short_url.GetRequestFromShortUrl(ctx, c, hashId)
+
+		dataTimestamp := data_timestamp.GetDataSourceTimestamp(ctx, c)
+		c.HTML(
+			http.StatusOK,
+			"index.html",
+			gin.H{
+				"title":                "Home Page",
```

Comment:
I'm not sure where this structure is defined, but it will be nice if you'll put all data structures in one place and use them here in other places where you call `gin.H`.

Reply:
Sorry Valentin, I did not understand which structure. If you mean `gin.H`, it is default `gin-gonic` structure to inject Controller data to html template.